src/main/java/com/faforever/client/game/GenerateMapController.java

```diff
@@ -0,0 +1,241 @@
+package com.faforever.client.game;
+
+import com.faforever.client.fx.Controller;
+import com.faforever.client.fx.DualStringListCell;
+import com.faforever.client.fx.JavaFxUtil;
+import com.faforever.client.map.MapBean;
+import com.faforever.client.map.generator.MapGeneratorService;
+import com.faforever.client.mod.FeaturedMod;
+import com.faforever.client.notification.NotificationService;
+import com.faforever.client.preferences.GeneratorPrefs;
+import com.faforever.client.preferences.PreferenceUpdateListener;
+import com.faforever.client.preferences.PreferencesService;
+import com.google.common.annotations.VisibleForTesting;
+import com.jfoenix.controls.JFXButton;
+import com.jfoenix.controls.JFXTextField;
+import javafx.application.Platform;
+import javafx.collections.FXCollections;
+import javafx.collections.transformation.FilteredList;
+import javafx.scene.control.CheckBox;
+import javafx.scene.control.MultipleSelectionModel;
+import javafx.scene.control.Spinner;
+import javafx.scene.control.Slider;
+import javafx.scene.control.CheckBox;
+import javafx.scene.control.SpinnerValueFactory.IntegerSpinnerValueFactory;
+import javafx.scene.input.KeyCode;
+import javafx.scene.layout.Pane;
+import javafx.util.StringConverter;
+import lombok.RequiredArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import java.lang.invoke.MethodHandles;
+import java.lang.ref.WeakReference;
+import java.util.Objects;
+import java.util.Random;
+import java.util.function.Function;
+
+@Component
+@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+@RequiredArgsConstructor
+public class GenerateMapController implements Controller<Pane> {
+
+  private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
```

Comment:
Use @Slf4j annotation instead

Reply:
I am unclear what using @slf4j should replace if I remove the logger then what should be used on line 227?

Reply:
https://projectlombok.org/api/lombok/extern/slf4j/Slf4j.html

Reply:
Done